### ISSUE TITLE
hyperv: EOF on ssh, exit status 1 for subsequent 'start' and 'delete' commands

### ISSUE THREAD

**Issue by mmoser18** [kubernetes/minikube#3732]
The installation of Minikube on my Windows 10 Pro system failed:
(this was all on an elevated cmd shell and I tried to follow this description: https://medium.com/jsonlovesyaml/12-step-tutorial-to-setup-kubernetes-on-your-windows-10-laptop-b7784b2253ce):

Prior to the below in HyperV's Virtual Switch Manager I had defined a virtual switch on the external network (my system's ethernet adapter) and called it  "minikube v switch" - as instructed.

First attempt:
```
C:\windows\system32>minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube v switch"
o   minikube v0.34.1 on windows (amd64)
>   Creating hyperv VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
@   Downloading Minikube ISO ...
 184.30 MB / 184.30 MB [============================================] 100.00% 0s
```
This process hung "forever" (10'?) until I finally aborted it (using ^C)
I then restarted it but it failed again:
```
C:\windows\system32>minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube v switch"
o   minikube v0.34.1 on windows (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Re-using the currently running hyperv VM for "minikube" ...
:   Waiting for SSH access ...
-   "minikube" IP address is 192.168.1.114
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
@   Downloading kubelet v1.13.3
@   Downloading kubeadm v1.13.3
!   Failed to update cluster: copy: pre-copy: command failed: sudo rm -f /lib/systemd/system/kubelet.service
stdout:
stderr: : EOF

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

The third attempt at first seemed to continue but then again hung "forever":
```
C:\windows\system32>minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube v switch"
o   minikube v0.34.1 on windows (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Re-using the currently running hyperv VM for "minikube" ...
:   Waiting for SSH access ...
-   "minikube" IP address is 192.168.1.114
-   Configuring Docker as the container runtime ...
-   Preparing Kubernetes environment ...
-   Pulling images required by Kubernetes v1.13.3 ...
:   Relaunching Kubernetes v1.13.3 using kubeadm ...
:   Waiting for kube-proxy to come back up ...
```
Again - I finally killed this after ~15' using ^C.
All further attempts to restart the installation all ended in:

```
C:\windows\system32>minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube v switch"
o   minikube v0.34.1 on windows (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Restarting existing hyperv VM for "minikube" ...
!   Unable to start VM: start: exit status 1

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

I also tried to revert, but that didn't work either:
```
C:\windows\system32>minikube delete
x   Deleting "minikube" from hyperv ...
!   Failed to delete cluster: exit status 1

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

Last attempt was to delete the created HyperV VM hoping this would restart the process from scratch but to no avail. All attempts since end in:
```
C:\windows\system32>minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube v switch"
o   minikube v0.34.1 on windows (amd64)
i   Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
:   Restarting existing hyperv VM for "minikube" ...
!   Unable to start VM: start: exit status 1

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

So, it seems I am between a hammer and a hard place and can't go forward nor backward. :-(
Any advice, how to clean out and/or reinstall would be a highly appreciated!

Regards,
Michael

**Comment by mmoser18** [comment]
I was able to "fix" this (well - at least I came out of the endless loop...) by removing the .minikube directory and starting over.

**Comment by tstromberg** [comment]
The initial errors indicate some sort of networking failure between the host and the VM - very strange.

The `minikube delete` errors are troublesome. If you can, do you mind sharing the output of:

`minikube delete --alsologtostderr -v=8`

I'd like to know why it's failing -- you shouldn't have to remove the .minikube directory.

**Comment by mmoser18** [comment]
Unfortunately I have deleted the directory after which - as I wrote - the problem vanished.

Sorry for not keeping the old directory. Stupid me!

 

If I encounter this issue again, I'll provide the requested data.

 

Regards,

Michael

**Comment by lucasgranberg** [comment]
Just for reference, I got this error when trying to start minikube over remote connection on windows without an admin powershell. Running as admin solved it. The -v8 got me the information I needed.

```
[stderr =====>] : Hyper-V\Remove-VM : You do not have the required permission to complete this task. Contact the administrator of the authorization policy for the computer 'OPSCOMPUTER'.
```

**Comment by yuxh** [comment]
similar problem on start and delete using win  10 and minikube 1.1, here is my log :
λ  minikube delete --alsologtostderr -v=8
W0605 08:14:50.107762   10988 root.go:151] Error reading config file at C:\Users\luke0\.minikube\config\config.json: open C:\Users\luke0\.minikube\config\config.json: The system cannot find the file specified.
I0605 08:14:50.113762   10988 notify.go:128] Checking for updates...
W0605 08:15:15.429763   10988 notify.go:57] Error getting json from minikube version url: error with http GET for endpoint https://storage.googleapis.com/minikube/releases.json: Get https://storage.googleapis.com/minikube/releases.json: net/http: TLS handshake timeout
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM minikube ).state
[stdout =====>] :
[stderr =====>] : Hyper-V\Get-VM : Hyper-V �Ҳ�������Ϊ��minikube������������
����λ�� ��:1 �ַ�: 3
+ ( Hyper-V\Get-VM minikube ).state
+   ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (minikube:String) [Get-VM]��VirtualizationException
    + FullyQualifiedErrorId : InvalidParameter,Microsoft.HyperV.PowerShell.Commands.GetVM


I0605 08:15:17.952765   10988 cluster.go:161] host is in state
* Deleting "minikube" from hyperv ...
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Hyper-V\Get-VM minikube ).state
[stdout =====>] :
[stderr =====>] : Hyper-V\Get-VM : Hyper-V �Ҳ�������Ϊ��minikube������������
����λ�� ��:1 �ַ�: 3
+ ( Hyper-V\Get-VM minikube ).state
+   ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (minikube:String) [Get-VM]��VirtualizationException
    + FullyQualifiedErrorId : InvalidParameter,Microsoft.HyperV.PowerShell.Commands.GetVM


[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Remove-VM minikube -Force
[stdout =====>] :
[stderr =====>] : Hyper-V\Remove-VM : Hyper-V �Ҳ�������Ϊ��minikube������������
����λ�� ��:1 �ַ�: 1
+ Hyper-V\Remove-VM minikube -Force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (:) [Remove-VM], VirtualizationException
    + FullyQualifiedErrorId : ObjectNotFound,Microsoft.HyperV.PowerShell.Commands.RemoveVM


W0605 08:15:22.907763   10988 exit.go:100] Failed to delete cluster: exit status 1

X Failed to delete cluster: exit status 1

**Comment by tstromberg** [comment]
@author: I believe this issue is now addressed by minikube v1.4, as it includes an updated Hyper-V driver. If you still see this issue with minikube v1.4 or higher, please reopen this issue by commenting with `/reopen`

Thank you for reporting this issue!
